src/commands/pulls/close.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @see: https://octokit.github.io/rest.js/#api-Pulls-update 
+ * There is no "closed" endpoint in the Octokit API. We use "update" with a `state` of "closed".
+ * const result = await octokit.pulls.update({owner, repo, number, title, body, state, base, maintainer_can_modify})
+ * /repos/:owner/:repo/pulls/:number
+ */
+const flow = require("lodash.flow")
+const fs = require("fs")
+
+const commonYargs = require("../../../lib/common-yargs")
+const printOutput = require("../../../lib/print-output")
+const authenticatedOctokit = require("../../../lib/octokit")
+
+/**
+ * yargs builder function.
+ *
+ * @param {import('yargs').Yargs} yargs - Instance of yargs
+ */
+const builder = yargs => {
+	const baseOptions = flow([
+		commonYargs.withToken,
+		commonYargs.withJson,
+		commonYargs.withBase,
+		commonYargs.withOwner,
+		commonYargs.withRepo,
+		commonYargs.withNumber,
+		commonYargs.withBody,
+		commonYargs.withTitle,
+	])
+	return baseOptions(yargs)
+		.option("maintainer_can_modify", {
+			describe: "Indicates whether maintainers can modify the pull request.",
+			type: "string",
+		})
+}
+
+/**
+ * Update a pull request.
+ *
+ * @param {object} argv - argv parsed and filtered by yargs
+ * @param {string} argv.token
+ * @param {string} argv.json
+ * @param {string} argv.owner
+ * @param {string} argv.repo
+ * @param {string} argv.number
+ * @param {string} [argv.title]
+ * @param {string} [argv.body]
+ * @param {string} [argv.base]
+ * @param {string} [argv.maintainer_can_modify]
+ * @throws {Error} - Throws an error if any required properties are invalid
+ */
+const handler = async ({ token, json, owner, repo, number, title, body, base, maintainer_can_modify }) => {
+
+	// Ensure that all required properties have values
+	const requiredProperties = {
+		owner,
+		repo,
+		number,
+	}
+	if (Object.values(requiredProperties).some(property => !property)) {
+		throw new Error(`Please provide all required properties: ${Object.keys(requiredProperties).join(", ")}`)
+	}
```

Comment:
This validation feels like a duplication of the validation that is provided by `yargs` when it enforces the options that you pass to `yargs.option()` and return via the `builder` method. Is there a case where the `handler` would be invoked directly without `process.argv` being validated and filtered by `yargs`?

We could add tests that ensure that the the `yargs` options are configured to validate things correctly:  https://github.com/yargs/yargs/blob/master/docs/advanced.md#testing-a-command-module - from that documentation it looks like the `parse` method will pass through error objects which we can make assertions against :grinning: